src/accessibility.ts

```diff
@@ -52,6 +55,12 @@ async function testScenario(elementOrWrapper: VNode | ReactWrapper) {
     // or "inapplicable" (no relevant HTML elements found).
     resultTypes: ['violations'],
   });
+
+  let violations = results.violations;
+  if (shouldIgnoreViolation) {
+    violations = violations.filter(v => !shouldIgnoreViolation(v));
```

Comment:
I suppose this may be working due to the overwriting of the `violations` reference, but perhaps we should make this function return `violations` instead of `result.violations`, otherwise it's not super obvious what this `if` statement does when overwriting the `violations` var.

Reply:
Yes, the function should return the filtered violations. That was a mistake on my part.

Reply:
As an aside, we currently lack tests for this module which would have made this more obvious.